src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -308,3 +311,177 @@ GetCitusDependedObjectArgs(int pgMetaTableVarno, int pgMetaTableOid)
 
 	return list_make2((Node *) metaTableOidConst, (Node *) oidVar);
 }
+
+
+/*
+ * DistOpsHasInvalidObject returns true if any address in the given node
+ * is invalid; otherwise, returns false. If ops is null or it has no
+ * implemented address method, we return false.
+ *
+ * We have some dist ops for which we should not validate.
+ * 1) We should not validate CREATE statements because no address exists
+ * here yet.
+ * 2) We should not validate '[DROP|ALTER] IF EXISTS' statements because it is ok
+ * by the semantics even if any object is invalid.
+ * 3) We should not validate 'ALTER ROLE ALL [SET|UNSET] because for the role ALL
+ * AlterRoleSetStmtObjectAddress returns an invalid address even though it should not.
+ */
+bool
+DistOpsHasInvalidObject(Node *node, const DistributeObjectOps *ops)
+{
+	if (ops && ops->operationType == DIST_OPS_CREATE)
+	{
```

Comment:
why we skip these statements? 

Please add comment on all statements where we return (e.g., move the comments in the function here. it is hard to maintain function comments as they change frequently)